### ISSUE TITLE
`__all__` operator do not work with multi layer subpackages

### ISSUE THREAD

**Issue by zyxkad** [python/mypy#15364]
**Bug Report**

(A clear and concise description of what the bug is.)

**To Reproduce**

File Tree:
<img width="237" alt="Screenshot 2023-06-03 at 10 32 33 PM" src="https://github.com/python/mypy/assets/91082411/44d04b4b-d77d-4bbd-a56a-addba7acb4c3">

`a/__init__.py`
```python
__all__ = []

from .aafunc import *
from . import aafunc
__all__.extend(aafunc.__all__)
```

`a/aafunc/__init__.py`
```python
__all__ = []

from .afunc import *
from . import afunc
__all__.extend(afunc.__all__)
```

`a/aafunc/afunc.py`
```python

__all__ = ['func']

def func() -> None:
	print('This is a func')
```

`b/__init__.py`
```python
from .main import main

__all__ = [
	'main'
]
```

`b/main.py`
```python
from a import afunc

__all__ = [
	'main',
]

def main():
	afunc()
```

**Expected Behavior**

```
Success: no issues found in 5 source files
```

**Actual Behavior**

```sh
$ mypy .
b/main.py:2: error: Module "a" has no attribute "afunc"; maybe "aafunc"?  [attr-defined]
Found 1 error in 1 file (checked 5 source files)
```

**Your Environment**

- Mypy version used: `mypy 1.3.0 (compiled: yes)`
- Mypy command-line flags: *None*
- Mypy configuration options from `mypy.ini` (and other config files): *None*
- Python version used: `Python 3.11.3`

**Comment by hauntsaninja** [comment]
Thanks, duplicate of https://github.com/python/mypy/issues/15300

**Comment by zyxkad** [comment]
Well, I mean, the operator works, but it only work with single layer subpackage, not multi layers @hauntsaninja
